mod-source-record-manager-server/src/main/java/org/folio/services/AbstractChunkProcessingService.java

```diff
@@ -56,6 +74,74 @@ public Future<Boolean> processChunk(RawRecordsDto incomingChunk, String jobExecu
         }).orElse(Future.failedFuture(new NotFoundException(String.format("Couldn't find JobExecution with id %s", jobExecutionId)))));
   }
 
+  private boolean isNotSupportedJobProfileExists(JobExecution jobExecution) {
+    boolean isExists = false;
+    List<ProfileSnapshotWrapper> childProfiles = jobExecution.getJobProfileSnapshotWrapper().getChildSnapshotWrappers();
+    if (!CollectionUtils.isEmpty(childProfiles)) {
+      isExists = isExistsMatchProfileToInstanceWithActionUpdateMarcBib(childProfiles);
+    }
+    return isExists;
+  }
+
+  //Disabled SONAR check "Loops with at most one iteration should be refactored" for recursive code
+  private boolean isExistsMatchProfileToInstanceWithActionUpdateMarcBib(Collection<ProfileSnapshotWrapper> profileSnapshotWrapperList) { //NOSONAR
+    for (ProfileSnapshotWrapper profileSnapshotWrapper : profileSnapshotWrapperList) {
+      if ((profileSnapshotWrapper.getContentType() == MATCH_PROFILE) && (isMatchingMarcBibToInstance(profileSnapshotWrapper))) {
+        ProfileSnapshotWrapper actionProfileSnapshotWrapper = getChildSnapshotWrapperByType(profileSnapshotWrapper, ACTION_PROFILE);
+        if (actionProfileSnapshotWrapper != null) {
+          ProfileSnapshotWrapper mappingProfileSnapshotWrapper = getChildSnapshotWrapperByType(actionProfileSnapshotWrapper, MAPPING_PROFILE);
+          if ((mappingProfileSnapshotWrapper != null) && (isMappingMarcBibToMarcBib(mappingProfileSnapshotWrapper))) {
+            return true;
+          }
+        }
+      }
+      if (!CollectionUtils.isEmpty(profileSnapshotWrapper.getChildSnapshotWrappers())) {
+        return isExistsMatchProfileToInstanceWithActionUpdateMarcBib(profileSnapshotWrapper.getChildSnapshotWrappers());
+      }
+    }
+    return false;
+  }
+
+  private ProfileSnapshotWrapper getChildSnapshotWrapperByType(ProfileSnapshotWrapper profileSnapshotWrapper,
+                                                                       ProfileSnapshotWrapper.ContentType contentType) {
+    if (!CollectionUtils.isEmpty(profileSnapshotWrapper.getChildSnapshotWrappers())) {
+      List<ProfileSnapshotWrapper> childSnapshotWrappers = profileSnapshotWrapper.getChildSnapshotWrappers();
+      for(ProfileSnapshotWrapper snapshotWrapper : childSnapshotWrappers) {
+        if (snapshotWrapper.getContentType() == contentType) {
+          return snapshotWrapper;
+        }
+      }
+    }
+    return null;
+  }
+
+  private boolean isMatchingMarcBibToInstance(ProfileSnapshotWrapper profileSnapshotWrapper) {
+    boolean result = false;
```

Comment:
@Aliaksandr-Fedasiuk  I think it would be better to rename "isMatched" or "matched" to represent a more logical variable.

Reply:
Ready.